molecule/header-auth-test/converge.yml

```diff
@@ -0,0 +1,149 @@
+- name: Tests
+  hosts: localhost
+  connection: local
+  vars:
+    custom_resource: "{{ lookup('template', cr_file_path) | from_yaml }}"
+  tasks:
+  - import_tasks: ../common/tasks.yml
+  - import_tasks: ../asserts/pod_asserts.yml
+
+  # Wait for Kiali to be running and accepting requests.
+  # This has the added benefit of confirming that we can access a
+  # Kiali endpoint that does not need authentication.
+  - import_tasks: ../common/wait_for_kiali_running.yml
+
+  - name: Make sure the test set the strategy to header
+    assert:
+      that:
+      - kiali_configmap.auth.strategy == "header"
+
+  - name: Assert that we can access Kiali console login screen that does not need authentication
+    uri:
+      url: "{{ kiali_base_url }}/console"
+      validate_certs: false
+
+  - name: Try to access Kiali api endpoint that requires authentication (should return error - we did not login yet)
+    uri:
+      url: "{{ kiali_base_url }}/api/namespaces"
+      status_code: 401
```

Comment:
This test fails because this is returning 200 when it should not. See: https://github.com/kiali/kiali/issues/3207#issuecomment-760590537

If you delete this line 28, and re-run the test, everything else will pass.

We should still merge this PR. This test *should* all pass. We need to fix the server-side code so this does return 401 (which is what this test is expecting).

Reply:
This problem should be fixed via this PR: https://github.com/mlbiam/kiali/pull/1
When I apply that fix, this molecule test all passes now.